week-four/app/src/main/java/pro/marvinhosea/movielist/repository/UserSharedPrefRepository.kt

```diff
@@ -0,0 +1,72 @@
+package pro.marvinhosea.movielist.repository
+
+import android.content.Context
+import androidx.preference.PreferenceManager
+
+object UserSharedPrefRepository : UserRepository {
+
+    private lateinit var applicationContext: Context
```

Comment:
Shouldn't be lateinit. You should pass in the SharedPreference object through the constructor.